lib/pure/times.nim

```diff
@@ -526,9 +554,58 @@ when not defined(JS):
     var cTimeInfo = timeInfo # for C++ we have to make a copy,
     # because the header of mktime is broken in my version of libc
     result = mktime(timeInfoToTM(cTimeInfo))
+    when defined(windows):
+      var secondOffset: TimeImpl = 0
+      if result == Time(-1) and cTimeInfo.year <= 1970:
```

Comment:
@flyx I know what assumptions you make, I want you to provide sources for these assumptions and add them as comments into the code.